app/models/support_interface/providers_filter.rb

```diff
@@ -4,59 +4,114 @@ class ProvidersFilter
+        :q,
+      ).presence || DEFAULT_STATE
     end
 
     def filters
-      onboarding_state_options = ONBOARDING_STAGES.map do |name, label|
-        {
-          value: name,
-          label: label,
-          checked: applied_filters[:onboarding_stages]&.include?(name),
-        }
-      end
-
       [
+        {
+          type: :search,
+          heading: 'Name or code',
+          value: applied_filters[:q],
+          name: 'q',
+        },
         {
           type: :checkboxes,
           heading: 'Onboarding stage',
-          options: onboarding_state_options,
+          options: hash_to_checkbox_options(:onboarding_stages, ONBOARDING_STAGES),
           name: 'onboarding_stages',
         },
         {
-          type: :search,
-          heading: 'Name or code',
-          value: applied_filters[:q],
-          name: 'q',
+          type: :checkboxes,
+          heading: 'Provider type',
+          options: hash_to_checkbox_options(:provider_types, PROVIDER_TYPES),
+          name: 'provider_types',
+        },
+        {
+          type: :checkboxes,
+          heading: 'Ratified by (only applies to providers with synced courses)',
+          options: hash_to_checkbox_options(:ratified_by, RATIFIED_BY),
+          name: 'ratified_by',
         },
       ]
     end
 
     def filter_records(providers)
-      if applied_filters[:q]
+      if applied_filters[:q].present?
         providers = providers.where("CONCAT(providers.name, ' ', providers.code) ILIKE ?", "%#{applied_filters[:q]}%")
+        @search_count = providers.count
+      else
+        @search_count = 0
       end
 
-      if applied_filters[:onboarding_stages]&.include?('synced')
-        providers = providers.where(sync_courses: true)
-      else
-        providers = providers.where(sync_courses: false)
+      unless applied_filters[:onboarding_stages]&.include?('any')
+        if applied_filters[:onboarding_stages]&.include?('synced')
+          providers = providers.where(sync_courses: true)
+        else
+          providers = providers.where(sync_courses: false)
+        end
+
+        if applied_filters[:onboarding_stages]&.include?('dsa_signed')
+          providers = providers.joins(:provider_agreements)
+        else
+          providers = providers.includes(:provider_agreements).where(provider_agreements: { provider_id: nil })
+        end
       end
 
-      if applied_filters[:onboarding_stages]&.include?('dsa_signed')
-        providers = providers.joins(:provider_agreements)
-      else
-        providers = providers.includes(:provider_agreements).where(provider_agreements: { provider_id: nil })
+      if applied_filters[:provider_types].present?
+        providers = providers.where(provider_type: applied_filters[:provider_types])
+      end
+
+      if applied_filters[:ratified_by].present?
+        ratifiers = Provider.where(provider_type: applied_filters[:ratified_by])
+        providers = providers.joins(:training_provider_permissions)
```

Comment:
This currently filters out anyone who falls into the category, but ratifies their own courses. I'm not sure how this is going to be used, is that intentional?

Reply:
As I understand it this shouldn't show any providers who aren't ratified by a separate accredited body.